File: frontend/src/services/authService.js
import { authApi } from '../api/authApi.js';

export const authService = {
  requestOtp: (email) => authApi.requestOtp(email),
  verifyOtp: (email, otp) => authApi.verifyOtp(email, otp),
  completeProfile: (payload) => authApi.completeProfile(payload),
  me: () => authApi.me(),
  logout: () => authApi.logout(),
};

File: backend/src/main/java/com/turfmanagement/service/AuthService.java
package com.turfmanagement.service;

import com.turfmanagement.dto.request.CompleteProfileDto;
import com.turfmanagement.dto.request.OtpRequestDto;
import com.turfmanagement.dto.request.OtpVerifyDto;
import com.turfmanagement.dto.response.AuthResponseDto;

public interface AuthService {

    void requestOtp(OtpRequestDto dto);

    AuthResponseDto verifyOtp(OtpVerifyDto dto);

    AuthResponseDto completeProfile(CompleteProfileDto dto);
}
